### DISCUSSION TITLE
Bad request when creating record

### DISCUSSION THREAD

**Discussion by jeansebastienvincent** [pocketbase/pocketbase#3802] [Q&A] [answered]
My app wants to create a record with following code 👍 
``        const supplierRecord =  await pb
          .collection("suppliers")
          .create(data, { fields: "id,name" });
``
In the devtools I get a 400 Bad Request error, but the record is created.
However, the fields asked are not returned.

In the pocketbase logs I find this :
{
  "errorMessage": "GoError: 501 Recipient syntax error at reflect.methodValueCall (native)"
}

Can anybody explain me what is happening and how I can solve it ( I am using v0.19.4).

**Comment by ganigeorgiev** [comment]
Do you have any custom Go or JS hooks code?

I'm not sure how to reproduce it. Could you provide a more detailed code sample and elaborate a little more how you are sending the request?

**Reply by ganigeorgiev** [reply, marked as answer]
The `501 Recipient...` error looks like a mailer error. If you have hooks that send custom emails make sure that the email addresses are properly formatted and existing.

**Reply by jeansebastienvincent** [reply]
Thanks. There is indeed a file with an 'onRecordAfterCreateRequest' in the pb_hooks folder.
I switched it out and I do not receive the error anymore.
For infomation  this is the code of the code 

```
onRecordAfterCreateRequest((e) => {

    const sentmailsCollection = $app
      .dao()
      .findCollectionByNameOrId("sentmails");

    const templates = arrayOf(
      new DynamicModel({
        id: "",
        documentName: "",
        content: "",
        documentType: "",
        language: "",
      })
    );

    $app
      .dao()
      .db()
      .newQuery(
        "SELECT * FROM standardDocument WHERE documentName = 'applicationReceived'"
      )
      .all(templates);

    const language = e.record.languageCode ?? "NL";
    console.log(language);
    console.log(e.record.name);
    console.log(e.record.email);
    console.log(e.record.id);

    const template = templates.find((t) => t.language === language);

    const subject =
      language === "EN"
        ? "Your application for xxx"
        : language === "FR"
        ? "Votre candidature pour xxx"
        : "Uw aanvraag voor xxx";

    const message = new MailerMessage({
      from: {
        address: $app.settings().meta.senderAddress,
        name: $app.settings().meta.senderName,
      },
      to: [{ name: e.record.name, address: e.record.email }],
      subject,
      html: `${template?.content}`,
    });

    $app.newMailClient().send(message);
  
}, "suppliers");
```

**Reply by ganigeorgiev** [reply]
`e.record` is a custom JS class converted from the Go code (aka. it is not POJO) and to access/modify its fields currently it is required to use the get/set methods defined in https://pocketbase.io/docs/js-records/#getset-record-fields (_the only exception to this are the base model fields like id, created, updated but for consistency they can also be accessed/modified via the get/set methods_).

In other words, you can try to replace the line:
```js
to: [{ name: e.record.name, address: e.record.email }],
```
with 
```js
to: [{ name: e.record.get("name"), address: e.record.get("email") }],
```
(_note that users by default are not required to have email and the above could fail if the record doesn't have an email; you can make the `email` field required from your auth collection settings > Options > "Always require email" toggle_)
